sd/etcd/registrar.go

```diff
@@ -19,9 +25,27 @@ type Registrar struct {
 type Service struct {
 	Key           string // unique key, e.g. "/service/foobar/1.2.3.4:8080"
 	Value         string // returned to subscribers, e.g. "http://1.2.3.4:8080"
+	TTL           *TTLOption
 	DeleteOptions *etcd.DeleteOptions
 }
 
+// TTLOption allow setting a key with a TTL, and regularly refreshes the lease with a goroutine
+type TTLOption struct {
+	Heartbeat time.Duration
+	TTL       time.Duration
+}
```

Comment:
Given that a TTLOption is constructed with a constructor, I don't think either of these fields needs to be exported. Is that correct?